### PR TITLE
[Merged by Bors] - chore: more byAsSorry backports

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Logic/Equiv/PartialEquiv.lean [hunk 1]
```diff
@@ -168,6 +168,7 @@ initialize_simps_projections PartialEquiv (toFun → apply, invFun → symm_appl
 -- @[simp, mfld_simps]
 -- theorem coe_mk (f : α → β) (g s t ml mr il ir) :
 --  (PartialEquiv.mk f g s t ml mr il ir : α → β) = f := by dsimp only
+-- #align local_equiv.coe_mk PartialEquiv.coe_mk
```
Comment (mattrobball) [inline Mathlib/Logic/Equiv/PartialEquiv.lean:171]: Did these sneak back in?

File: Mathlib/NumberTheory/Cyclotomic/Rat.lean [hunk 2]
```diff
@@ -265,7 +267,7 @@ noncomputable def subOneIntegralPowerBasis [IsCyclotomicExtension {p ^ k} ℚ K]
       · exact Subalgebra.sub_mem _ (hζ.isIntegral (by simp)) (Subalgebra.one_mem _))
 
 @[simp]
-theorem subOneIntegralPowerBasis_gen [IsCyclotomicExtension {p ^ k} ℚ K]
+theorem subOneIntegralPowerBasis_gen [hcycl : IsCyclotomicExtension {p ^ k} ℚ K]
```
Comment (mattrobball) [inline Mathlib/NumberTheory/Cyclotomic/Rat.lean:270]: Why do we have to name the instance here?

File: Mathlib/RingTheory/AdicCompletion/Exactness.lean [hunk 3]
```diff
@@ -39,31 +39,33 @@ section Surjectivity
...
   | .succ n =>
       let y := (hf (x (n + 1))).choose
       have hy := (hf (x (n + 1))).choose_spec
-      let ⟨yₙ, (hyₙ : f yₙ = x n)⟩ := mapPreimage x n
+      let ⟨yₙ, (hyₙ : f yₙ = x n)⟩ := mapPreimage hf x n
       let ⟨⟨d, _⟩, (p : f d = f (yₙ - y))⟩ := mapPreimageDelta hf x hy hyₙ
       ⟨yₙ - d, by simpa [p]⟩
 
+variable (hf : Function.Surjective f)
+include hf
```
Comment (mattrobball) [inline Mathlib/RingTheory/AdicCompletion/Exactness.lean:67]: Is this just needed in `map_surjective`?

File: Mathlib/Data/Set/Function.lean [hunk 4]
```diff
@@ -1138,7 +1138,7 @@ attribute [local instance] Classical.propDecidable
 
 /-- Construct the inverse for a function `f` on domain `s`. This function is a right inverse of `f`
 on `f '' s`. For a computable version, see `Function.Embedding.invOfMemRange`. -/
-noncomputable def invFunOn (f : α → β) (s : Set α) (b : β) : α :=
+noncomputable def invFunOn [Nonempty α] (f : α → β) (s : Set α) (b : β) : α :=
```
Comment (mattrobball) [inline Mathlib/Data/Set/Function.lean:1141]: Maybe delete the `Nonempty` above and move it below `invFunOn`?

File: Mathlib/Algebra/Category/ModuleCat/Differentials/Presheaf.lean [hunk 5]
```diff
@@ -149,7 +149,8 @@ variable (d : ∀ (X : Dᵒᵖ), (M.obj X).Derivation (φ'.app X))
 /-- Given a morphism of presheaves of commutative rings `φ'`, this is the
 in derivation `M.Derivation' φ'` that is given by a compatible family of derivations
 with values in the modules `M.obj X` for all `X`. -/
-def mk : M.Derivation' φ' where
+def mk (d_map : ∀ ⦃X Y : Dᵒᵖ⦄ (f : X ⟶ Y) (x : R.obj X),
```
Comment (mattrobball) [inline Mathlib/Algebra/Category/ModuleCat/Differentials/Presheaf.lean:152]: Should some of the variables move down below this?

File: Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafify.lean [hunk 6]
```diff
@@ -368,7 +368,7 @@ variable {M₀' : PresheafOfModules.{v} R₀} {A' : Sheaf J AddCommGrp.{v}}
 induced by morphisms `τ₀ : M₀ ⟶ M₀'` and `τ : A ⟶ A'`
 which satisfy `τ₀.hom ≫ φ' = φ ≫ τ.val`. -/
 @[simps]
-def sheafifyMap : sheafify α φ ⟶ sheafify α φ' where
+def sheafifyMap (fac : τ₀.hom ≫ φ' = φ ≫ τ.val) : sheafify α φ ⟶ sheafify α φ' where
```
Comment (mattrobball) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafify.lean:371]: Should `fac` in the variable block above move down?

Comment (semorrison) [reply Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafify.lean:371]: In fact, just removed.

File: Mathlib/NumberTheory/Cyclotomic/Rat.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/NumberTheory/Cyclotomic/Rat.lean]: It was used in a named argument a few lines below, but I've cleaned up and use a `have` statement for the local instance now.

**Comment by semorrison** [comment]
Thanks @mattrobball, I've resolved your helpful suggestions.

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**